### PR TITLE
Rabbitmq driver

### PR DESCRIPTION
**Pull request by Raving-hash**
I hava finished the driver of rabbitmq. It conclued producer and consumer.

### CODE REVIEW

File: driver-rabbitmq/src/main/java/io/openchaos/driver/rabbitmq/RabbitMQDriver.java [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+        this.rmqClientConfig = readConfigForClient(configurationFile);
+        this.rmqBrokerConfig = readConfigForBroker(configurationFile);
+        this.rmqConfig = readConfigForRMQ(configurationFile);
+        this.nodes = nodes;
+        this.user = rmqClientConfig.user;
+        this.password = rmqClientConfig.password;
+        sync = new Sync(nodes);
+    }
+
+    public void intialState() {
```
Comment (HScarb) [inline driver-rabbitmq/src/main/java/io/openchaos/driver/rabbitmq/RabbitMQDriver.java:99]: Spelling mistake `initialState`

File: driver-rabbitmq/src/main/java/io/openchaos/driver/rabbitmq/RabbitMQDriver.java [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+    private ConnectionFactory factory;
+    private RabbitMQChaosState state;
+    private GenericObjectPool<Channel> stateChannelPool;
+    private ObjectPool<Channel> producerChannelPool;
+    private GenericObjectPool<Channel> consumerChannelPool;
+    private Connection consumerConnection;
+    private Connection producerConnection;
+    private Connection stateConnection;
+    private Sync sync;
+    private volatile String curState = "stop";
```
Comment (HScarb) [inline driver-rabbitmq/src/main/java/io/openchaos/driver/rabbitmq/RabbitMQDriver.java:63]: It's better to use enum as a state

File: driver-rabbitmq/src/main/java/io/openchaos/driver/rabbitmq/RabbitMQChaosState.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                String res = SshUtil.execCommandWithArgsReturnStr(leader, "rabbitmq-queues quorum_status \"openchaos_client_1\" | grep leader ");
+                String[] s = res.split(" ");
+                leaderAddr.add(getHost(s[1]));
+            } catch (Exception e) {
+                log.warn("Get leader failed!");
+            }
+        } else if (haMode == HaMode.classic) {
+            String command = "curl -s -u " + user + ":" + password + " http://" + leader + ":15672/api/queues/%2f/openchaos_client_1 | python -m json.tool | grep node";
+            try {
+                String s = SshUtil.execCommandWithArgsReturnStr(leader, command);
```
Comment (HScarb) [inline driver-rabbitmq/src/main/java/io/openchaos/driver/rabbitmq/RabbitMQChaosState.java:58]: You can use httpclient to call rabbitmq API to get leader node

File: driver-rabbitmq/src/main/java/io/openchaos/driver/rabbitmq/RabbitMQDriver.java [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+            tmpChan.queueDelete(queueName);
+            tmpChan.queueDeclare(queueName, false, false, false, null);
+            tmpChan.close();
+            tmpCon.close();
+        } catch (IOException | TimeoutException e) {
+            throw new RuntimeException(e);
+        }
+        state = new RabbitMQChaosState("", queueName, state.getLeader().iterator().next(), rmqBrokerConfig.haMode, user, password);
+        factory = new ConnectionFactory();
+        factory.setHost(state.getLeader().iterator().next());
```
Comment (HScarb) [inline driver-rabbitmq/src/main/java/io/openchaos/driver/rabbitmq/RabbitMQDriver.java:125]: If I didn't set address in host file for the leader rabbitmq node, an exception will be thrown here

Comment (Raving-hash) [reply driver-rabbitmq/src/main/java/io/openchaos/driver/rabbitmq/RabbitMQDriver.java:125]: Thus users must set all address in host file before they run openchaos.

**Pull request merged**